server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/step/ViewsPersistComponentsStepTest.java

```diff
@@ -348,6 +352,75 @@ public void update_copy_component_uuid_of_sub_view() {
     assertThat(subViewReloaded.getCopyResourceUuid()).isEqualTo("NEW_COPY");
   }
 
+  @Test
+  public void persists_new_components_as_private_if_root_does_not_exist_yet_out_of_functional_transaction() {
+    ComponentDto project = dbTester.components().insertComponent(newProjectDto(dbTester.organizations().insert()));
+    treeRootHolder.setRoot(
+      createViewBuilder()
+        .addChildren(
+          createSubView1Builder(null)
+            .addChildren(
+              createProjectView1Builder(project, null).build())
+            .build())
+        .build());
+
+    underTest.execute();
+
+    Stream.of(VIEW_UUID, SUBVIEW_1_UUID, PROJECT_VIEW_1_UUID)
+      .forEach(uuid -> assertThat(dbClient.componentDao().selectByUuid(dbTester.getSession(), uuid).get().isPrivate()).isFalse());
+  }
+
+  @Test
+  public void persists_new_components_with_private_flag_from_root_in_db_out_of_functional_transaction() {
+    boolean isRootPrivate = new Random().nextBoolean();
```

Comment:
I'll change the name of the method, but to this test, whether the private flag of the root is true or not doesn't matter, what matters is that the new components get the same value.
using a random value emphasis that the actual value doesn't matter